core/src/executor/evaluate/function.rs

```diff
@@ -58,17 +58,27 @@ macro_rules! eval_to_float {
 // --- text ---
 
 pub fn concat(exprs: Vec<Evaluated<'_>>) -> Result<Evaluated> {
+    let expr_type: Value = match exprs.first() {
```

Comment:
```rust
    enum BreakCase {
        Null,
        Err(crate::result::Error),
    }

    let control_flow = exprs.into_iter().map(|expr| expr.try_into()).try_fold(
        None,
        |left: Option<Value>, right: Result<Value>| match right {
            Ok(value) if value.is_null() => ControlFlow::Break(BreakCase::Null),
            Err(err) => ControlFlow::Break(BreakCase::Err(err)),
            Ok(value) => ControlFlow::Continue(match left {
                Some(left) => Some(left.concat(value)),
                None => Some(value),
            }),
        },
    );

    match control_flow {
        ControlFlow::Continue(value) => {
            let value = value.ok_or_else(|| ValueError::EmptyArgNotAllowedInConcat)?;

            Ok(Evaluated::from(value))
        }
        ControlFlow::Break(BreakCase::Null) => Ok(Evaluated::from(Value::Null)),
        ControlFlow::Break(BreakCase::Err(err)) => Err(err),
    }
```
Writing like this makes `Clone::clone` disappear.

Reply:
check this @panarch

Reply:
Amazing